lib/serviceBuilder.js

```diff
@@ -102,8 +102,16 @@ function makeCallAsBuffer(endUrl, method, body, options) {
       }
       // ugly
       res.payload = data
+
       if (isStatusCodeNotAllowed(options, res)) {
-        return reject(statusCodeError(res.statusCode, options.allowedStatusCodes, JSON.parse(data).message))
+        let errorMessage
+
+        try {
+          errorMessage = JSON.parse(data).message
+        // eslint-disable-next-line no-empty
+        } catch (exc) {}
```

Comment:
What about a warn/debug log instead of leaving an empty catch block?

Reply:
@fredmaggiowski 
But what about the message field? Without parsing data you would lose `message` field and the method `statusCodeError` will always receive an empty string or undefined

Reply:
I agree with you for the empty catch statement, however logging inside the catch would mean passing a logger as additional parameter to the `makeCallAsBuffer` function.
I don't know if this could be the right approach

Reply:
ok, there's even a better solution then:

Can we verify that the response content type and perform the JSON.parse only on applicable content types (e.g. application/json)?